nkms/characters.py

```diff
@@ -173,31 +162,32 @@ def encrypt_for(self,
         :return: A tuple, (ciphertext, signature).  If sign==False,
             then signature will be NOT_SIGNED.
         """
-        actor = self._lookup_actor(recipient)
-
+        recipient_pubkey_enc = recipient.public_key(EncryptingPower)
         if sign:
             if sign_plaintext:
-                signature = self.seal(plaintext)
-                message_kit = self._crypto_power.encrypt_for(
-                    actor.public_key(EncryptingPower), signature + plaintext)
+                # Sign first, encrypt second.
+                signature = self.stamp(plaintext)
+                ciphertext, capsule = pre.encrypt(recipient_pubkey_enc, signature + plaintext)
             else:
-                message_kit = self._crypto_power.encrypt_for(
-                        actor.public_key(EncryptingPower), plaintext)
-                signature = self.seal(message_kit.ciphertext)
-            message_kit.alice_pubkey = self.public_key(SigningPower)
+                # Encrypt first, sign second.
+                ciphertext, capsule = pre.encrypt(recipient_pubkey_enc, plaintext)
+                signature = self.stamp(ciphertext)
```

Comment:
This is great

Reply:
Still would prefer to perform this as a `CryptoKit` though.

Reply:
It is - just a few lines down I make a `MessageKit` from it.  Or do you mean that you prefer for `pre.encrypt` to return the `MessageKit`?

Reply:
I meant I prefer the `stamp` to be a kit instead of a `stamp`

Reply:
Oh right, I had forgotten that you had that notion.  I still don't quite get it - what goes in this kit?